pygfx/utils/trackable.py

```diff
@@ -150,6 +149,8 @@ def __getattribute__(self, key):
             value = self[key]
             return value
         except KeyError:
+            if hasattr(dict, key):
+                return dict.__getattribute__(self, key)
```

Comment:
I feel like a getattr with a try/except clause would be more performant, to avoid looking up the key twice?

Reply:
I tried the following:
```python
try:
    value = self[key]
    return value
except KeyError:
    return dict.__getattribute__(self, key)
```
and the happy path seems ever so slightly faster. When the attribute is not there, it is a bit slower, but that code path should not be hit too often. Certainly not within regular pygfx usage.

Reply:
No, I mean, the code path raises `AttributeError(key)` anyway, so why not just:

```py
    def __getattribute__(self, key):
        if key.startswith("__"):
            return dict.__getattribute__(self, key)
        value = undefined
        try:
            value = self[key]
            return value
        except KeyError:
            return dict.__getattribute__(self, key)
        finally:
            if global_context:
                global_context.tracker._track_get(
                    self, key, value, global_context.label
                )
```